community/server/src/test/java/org/neo4j/server/integration/StartupLoggingIT.java

```diff
@@ -0,0 +1,124 @@
+            boot.stop();
+        }
+
+        // Then
+        List<String> captured = stdout.lines();
+        assertThat( captured, matchesLines(
+                info( "Neo4j is starting.." ),
+                info( "Neo4j has started! The HTTP API and the Neo4j Browser are now available at http://.*:7474/" ),
+
+                info( "Neo4j is stopping.." ),
+                info( "Neo4j has stopped, see you next time!" )
+        ) );
+    }
+
+    public static Matcher<List<String>> matchesLines( final Matcher<String> ... lineMatchers )
+    {
+        return new TypeSafeMatcher<List<String>>()
+        {
+            @Override
+            protected boolean matchesSafely( List<String> lines )
+            {
+                if(lineMatchers.length != lines.size())
+                {
+                    return false;
+                }
+
+                for ( int i = 0; i < lines.size(); i++ )
+                {
+                    if( !lineMatchers[i].matches( lines.get( i ) ) )
+                    {
+                        return false;
+                    }
+                }
+                return true;
+            }
+
+            @Override
+            public void describeTo( Description description )
+            {
+                description.appendList( "", "\n", "", asList(lineMatchers) );
+            }
+        };
+    }
+
+    public static Matcher<String> warn( String messagePattern ) { return line("WARN", messagePattern); }
+    public static Matcher<String> info( String messagePattern ) { return line("INFO", messagePattern); }
+
+    public static Matcher<String> line( final String level, final String messagePattern )
+    {
+        return new TypeSafeMatcher<String>()
+        {
+            @Override
+            protected boolean matchesSafely( String line )
+            {
+                // eg. 2015-10-27 15:44:18.049-0500 INFO  Successfully started database
+                return line.matches( ".*" + level + "\\t" + messagePattern );
+            }
+
+            @Override
+            public void describeTo( Description description )
+            {
+                description.appendText( level ).appendText( " " ).appendText( messagePattern );
+            }
+        };
+    }
+
+    public static class CaptureStdOut
+    {
+        private final ByteArrayOutputStream captured = new ByteArrayOutputStream();
+
+        public AutoCloseable capture()
+        {
+            final PrintStream original = System.out;
+            System.setOut( new PrintStream( captured ) );
+            return new AutoCloseable()
+            {
+                @Override
+                public void close() throws Exception
+                {
+                    System.setOut( original );
+                }
+            };
+        }
+
+        public List<String> lines() throws UnsupportedEncodingException
+        {
+            String s = this.captured.toString( "UTF-8" );
```

Comment:
I would, but toString on `ByteArrayOutputStream` only takes a string charset :( I don't think it's a huge issue here, since this is in a test, so there's no catching it anywhere anyway.

I did notice that i missed a system.out on the next line tho..